README_Develop_with_devenv.md

```diff
@@ -0,0 +1,160 @@
+
+## [devenv](https://deven.sh/) can be use to create an isolated development environment.
+
+Here are a set of instructions to setup a development environment that is more interactive than
+Docker containers while being isolated. It requires to install a Nix system on top of your current
+Linux environment.
+
+
+### Step 1: Install `devenv`
+
+Follow the instructions from [devenv - Getting Started](https://deven.shgetting-started//):
+
+- Install a base Nix system:
+        `sh <(curl -L https://nixos.org/nix/install) --daemon`
+
+- Install _Cachix_ which provides pre-compiled binaries (especially useful when installing
+  Python packages):
+        `nix-env -iA cachix -f https://cachix.org/api/v1/install`
+        `cachix use devenv`
+        `cachix use nixpkgs-python`
+
+- Install `devenv`:
+        `nix-env -if https://github.com/cachix/devenv/tarball/latest`
+
+### Step 2: Prepare the environment
+
+- We will use a local home for all Cargo crates: `mkdir .cargo_home`
+
+- Within the top directory of the PostgresML repo, run `devenv init`.
+
+- Create or replace a file named `devenv.nix` containing
+
+```shell
+{ pkgs, ... }:
+
+{
+  # https://devenv.sh/basics/
+  env.GREET = "devenv";
+
+  # https://devenv.sh/packages/
+  packages = [
+    pkgs.git
+    pkgs.postgresql_15
+
+    pkgs.bison
+    pkgs.flex
+    pkgs.cmake
+    pkgs.gcc
+
+    pkgs.openssl
+
+    pkgs.llvmPackages.openmp
+    pkgs.lightgbm
+    pkgs.xgboost
+
+    # Deepspeed requires either one.
+    pkgs.cudaPackages.cuda_nvcc
+    # pkgs.hipcc
+    ];
+
+    # https://devenv.sh/scripts/
+    scripts.hello.exec = "\necho Starting with $GREET\n";
+
+    enterShell = ''
+    hello
+    '';
+
+    # https://devenv.sh/languages/
+    languages.nix.enable = true;
+
+    languages.python = {
+        enable = true;
+        version = "3.10";
+
+        venv.enable = true;
+
+        # Copied from requirements.txt
+        # Removed because unneeded
+        #   accelerate==0.19.0
+        #   sacremoses==0.0.53
+        #   einops==0.6.1
```

Comment:
These dependencies are required for the latest models downloaded from huggingface... huggingface & python dependency mgmt is a bit of a mess, it's a try it and see what works and then what you need crapshoot... These dependencies are also changing on something like a weekly cadence, so maintaining duplicate lists, rather than a shared requirements.txt is likely to cause issues when one is updated but not the other.

Reply:
I noticed as well and completely agree. I imagine having a script automatically building a new devenv.nix before switching into the environment. 

But I want to get to the point of a successfull compilation before doing that (the compilation problem is not related to bad Python imports).